### DISCUSSION TITLE
[Feature] More SwitchPresenter Samples

### DISCUSSION THREAD

**Discussion by hawkerm** [CommunityToolkit/Windows#401] [Ideas]
### Describe the problem

`SwitchPresenter` is powerful, but there are complexities when trying to use it with certain types.

### Describe the solution

- Show using `{x:Null}` as a case (I know this is working, at least with a nullable `?` type).

- Show casing on an integer (I don't know if this is working?).

- Using with a `NavigationView` for page selection vs. Frame

- Not sure of other potential scenarios?

### Help us help you

Yes, I'd like to be assigned to work on this item.

**Comment by ghost** [comment]
Hello, 'hawkerm! Thanks for submitting a new feature request. I've automatically added a vote 👍 reaction to help get things started. Other community members can vote to help us prioritize this feature in the future!

**Comment by ThomasDallmeier** [comment]
An example with animations for the switch content would be nice

**Comment by kg-github** [comment]
On a similar note, the docs show simple matches for strings and enums but how do you specify a complex type as a string in xaml for matching a case? 

E.g. If you have a ComboBox displaying a list of a complex type, the ComboBox is using ToString() of the type but when the SwitchPresenter is binding to the SelectedItem of the ComboBox, it is getting the object itself so then how to specify as a string in the case for matching?

**Reply by kg-github** [reply]
I'll answer my own question for anyone else that has the same issue: you need to implement an IValueConverter for the binder. This doc helped:
https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.data.binding.converter?view=winrt-22621

**Comment by michael-hawker** [comment]
Moved over to the new repo. I know @niels9001 just added a Loading/Animation sample, so we have done a bit of this. I was realizing that it could be handy to just have a base Boolean one as well. Will see if I can find an afternoon to try and create a few and improve stuff, as this is a powerful tool that can be used in a variety of ways.
